consult-search.el

```diff
@@ -0,0 +1,92 @@
+;;; consult-search.el --- Providing search via Consult and Projectile  -*- lexical-binding: t; -*-
+
+;; Author: Daniel Mendler, Consult and Selectrum contributors
+;; Maintainer: Daniel Mendler
+;; Created: 2020
+;; License: GPL-3.0-or-later
+;; Version: 0.1
+;; Package-Requires: ((projectile "2.0") (emacs "26.1"))
+;; Homepage: https://github.com/minad/consult
+
+;; This file is not part of GNU Emacs.
+
+;; This program is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; consult-search integrates command line search tools and presents the results
+;; using completing-read.  This was written to work with consult
+
+;; This only works with ag for now
+
+;;; Code:
+
+(require 'projectile)
```

Comment:
It would be great if we can get rid of this dependency and offer instead a consult-project-root-function defcustom such that the user can either configure project.el or projectile.el from outside.